posts/2023-12-31-foss-engines-overview.md

```diff
@@ -0,0 +1,112 @@
+
+Asset libraries provide a wealth of resources that empower developers to learn the tools quickly and get their game to a playable state. The community assembled a library of assets available on the official website for the community to use and share. The library includes tutorials, plugins, and templates that address subjects like physics, networking, and input management. There are even entire games and applications available in the asset library to either build on or use as a reference while learning the engine. Bevy’s structure encourages developers to use any of the resources from this library freely as part of the building blocks that will make up their game. In conjunction with Rust’s package manager, there is a strong emphasis on modularity at every level of the engine. 
+
+
+### Godot
+* Written in C++, Supported Languages: GDScript, C#
+* Platforms:
+    * Development: Windows, Linux, Mac
+    * Target: Windows, Linux, Mac, iOS, Android, WASM, HTML5
+
+Among all the modern FOSS game engines available to date, Godot has the largest and most active community. As the drama around Unity has unfolded, you have more than likely heard mentions of Godot on more than a few occasions. It is not without merit, as Godot encourages developers to shape the engine around their needs. Coming in at only 40 MB, the engine includes a lightweight, multi-platform editor capable of running on any major operating system (Windows, Mac, and Linux). In fact, you can even use a web-based version, albeit with some constraints. Godot can meet developers on whatever platform works best for them.
+
+GDScript is Godot’s primary programming language. While the prospect of learning an engine-specific language may turn you off at first, don’t fret! It shares a lot of commonalities with Python and Godot provides detailed documentation on how to use the language. Assuming you already have some experience with object-oriented programming, it won’t take long to get going with GDScript. You can even also use C# for scripting if that is more up your alley, as it’s the other language officially supported by Godot. That said, if you would still like to write some code in another language entirely, Godot provides the means to use alternative programming languages by the way of GDExtension.
+
+GDExtension theoretically supports any programming language that can interact with its C-based API. While Godot officially supports scripting in GDScript and C#, the system allows the community to introduce new language bindings to Godot’s development ecosystem, including Rust, Python, Lua, and Java. Not all language bindings are feature complete, but many are in active development. With that in mind, it’s unnecessary to commit to one language for an entire project, as GDExtension languages can work alongside GDScript. This means developers can, for example, even use GDScript with other languages in the same project.
```

Comment:
The list could include Swift! That language bindings is currently handled by Miguel de Icaza, of GNOME and Mono fame.

Reply:
Added